api/org/commcare/api/persistence/SqliteIndexedStorageUtility.java

```diff
@@ -37,12 +37,23 @@
     private final Class<T> prototype;
     private final String tableName;
     private final String sandboxId;
+    private final File databaseFolder;
 
-    public SqliteIndexedStorageUtility(Class<T> prototype, String sandboxId, String tableName) {
+    public SqliteIndexedStorageUtility(Class<T> prototype, String sandboxId, String tableName){
+        this(prototype, sandboxId, tableName, null);
+    }
+
+    public SqliteIndexedStorageUtility(Class<T> prototype, String sandboxId, String tableName, String databasePath) {
         this.tableName = tableName;
         this.sandboxId = sandboxId;
         this.prototype = prototype;
 
+        if(databasePath == null){
+            databaseFolder = new File(DATABASE_FOLDER);
```

Comment:
I'm going to double down on that, actually, because it looks like we're basically doing this anyway, which means some of these code paths are never triggered.